app/build.gradle

```diff
@@ -121,6 +121,7 @@ dependencies {
     // Dagger
     implementation "com.google.dagger:dagger:$dagger_version"
     implementation "com.google.dagger:dagger-android-support:$dagger_version"
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```

Comment:
Such as dagger and other dependencies move version to ext part. Moreover, keep dagger dependencies together and move androidx with the other androidx dependencies.